crates/bevy_reflect/src/impls/std.rs

```diff
@@ -90,7 +90,26 @@ impl_from_reflect_value!(NonZeroU16);
 impl_from_reflect_value!(NonZeroU8);
 impl_from_reflect_value!(NonZeroI8);
 
-impl<T: FromReflect> Array for Vec<T> {
+impl_type_name!(Vec<T: TypeName + 'static>);
+impl_type_name!(HashMap<K: TypeName + 'static, V: TypeName + 'static>);
+impl_type_name!(Option<T: TypeName + 'static>);
+
+// impl_type_name expecte a type name followed by generic between `<` and `>`.
+// so array is manually implemented.
+impl<T: TypeName + 'static, const N: usize> TypeName for [T; N] {
+    fn name() -> &'static str {
+        static CELL: GenericTypeNameCell = GenericTypeNameCell::new();
+        CELL.get_or_insert::<Self, _>(|| format!("[{}; {N}]", T::name()))
+    }
+}
+
+impl TypeName for Cow<'static, str> {
```

Comment:
Can we not use `impl_type_name` for this too? Or does the lifetime mess it up?

Reply:
The lifetime mess it up.